ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -177,15 +177,35 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			rayJobInstance.Status.DashboardURL = clientURL
 		}
 
+		if rayJobInstance.Spec.SubmissionMode == rayv1.UserMode {
+			logger.Info("SubmissionMode is UserMode and the RayCluster is created. Transition the status from `Initializing` to `Waiting`.")
+			rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusWaiting
+			break
+		}
+
 		if rayJobInstance.Spec.SubmissionMode == rayv1.K8sJobMode {
 			if err := r.createK8sJobIfNeed(ctx, rayJobInstance, rayClusterInstance); err != nil {
 				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 			}
 		}
 
-		logger.Info("Both RayCluster and the submitter K8s Job are created. Transition the status from `Initializing` to `Running`.",
+		logger.Info("SubmissionMode is K8sJobMode and both RayCluster and the submitter K8s Job are created. Transition the status from `Initializing` to `Running`.",
```

Comment:
Is it possible for HTTPMode going through here?

Reply:
https://docs.ray.io/en/latest/cluster/kubernetes/getting-started/rayjob-quick-start.html

Reply:
https://github.com/ray-project/kuberay/blob/d0257929a30a710b3610adf739130573f6d43717/ray-operator/config/samples/ray-job.sample.yaml#L8

Reply:
You are right. I tested it by setting breakpoint in IDE.